toolkit/imageconfigs/raid.json

```diff
@@ -0,0 +1,131 @@
+{
+    "Disks": [
+        {
+            "PartitionTableType": "gpt",
+            "MaxSize": 5300,
+            "Artifacts": [
+                {
+                    "Name": "coreA",
+                    "Type": "vhdx"
+                }
+            ],
+            "Partitions": [
+                {
+                    "ID": "efi",
+                    "Flags": [
+                        "esp",
+                        "boot"
+                    ],
+                    "Start": 1,
+                    "End": 9,
+                    "FsType": "fat32"
+                },
+                {
+                    "ID": "boot_1",
+                    "Start": 10,
+                    "End": 410
+                },
+                {
+                    "ID": "boot_2",
+                    "Start": 410,
+                    "End": 810
+                },
+                {
+                    "ID": "boot_3",
+                    "Start": 810,
+                    "End": 1210
+                },
+                {
+                    "ID": "rootfs_1",
+                    "Start": 1210,
+                    "End": 3210
+                },
+                {
+                    "ID": "rootfs_2",
+                    "Start": 3210,
+                    "End": 4210
+                },
+                {
+                    "ID": "rootfs_3",
+                    "Start": 4210,
+                    "End": 5210
+                }
+            ]
+        },
+        {
+            "ID": "raidBoot",
+            "TargetDisk": {
+                "Type": "raid",
+                "RaidConfig": {
+                    "ComponentPartIDs": [
+                        "boot_1",
+                        "boot_2",
+                        "boot_3"
+                    ],
+                    "Level": 5,
+                    "RaidID": "boot"
+                }
+            },
+            "Partitions": [
+                {
+                    "ID": "boot",
+                    "FsType": "ext4"
+                }
+            ]
+        },
+        {
+            "TargetDisk": {
+                "Type": "raid",
+                "RaidConfig": {
+                    "ComponentPartIDs": [
+                        "rootfs_1",
+                        "rootfs_2",
+                        "rootfs_3"
+                    ],
+                    "Level": 5,
```

Comment:
I am pretty sure if we set a value other than raid 1, or 0, ie anything higher than raid 4 for rootfs, I THINK you need to include the ko raid456 into the initrd. ie mkinitrd --with=raid456  ...

Reply:
The new grub `mdraid1x` module will handle this. Adding it in PR #5099 .